app/middlewares/session.js

```diff
@@ -0,0 +1,28 @@
+const bcrypt = require('bcrypt');
+
+const { findUserByEmail } = require('../services/user');
+const { sessionError } = require('../errors');
+
+const findUser = (request, _response, next) => {
+  // eslint-disable-next-line consistent-return
+  findUserByEmail(request.body.email).then(user => {
+    if (!user) return next(sessionError('User not found'));
+    request.user = user;
+    next();
+  });
+};
+
+const comparePassword = (request, _response, next) => {
+  const givenPassword = request.body.password;
+  const { user } = request;
+
+  bcrypt.compare(givenPassword, user.password, (err, res) => {
```

Comment:
always try to avoid using callbacks and try to promisify callback based functions instead. This is mainly because in refactorings having everything async being either async/await or promise based helps to avoid unexpected behaviours